### PR TITLE
[FIX] stock_account: fix invoice_state on create stock picking

### PR DESCRIPTION
**Pull request by acysos**
Hello,

When a sale is confirmed check first procurement.rule_id.invoice_state is this is none, don't check procurement.invoice_state. So, the stock.picking invoice_state is not aplicable.

Greetings

### CODE REVIEW

File: addons/stock_account/stock.py [hunk 1]
```diff
@@ -69,7 +69,12 @@ class procurement_order(osv.osv):
 
     def _run_move_create(self, cr, uid, procurement, context=None):
         res = super(procurement_order, self)._run_move_create(cr, uid, procurement, context=context)
-        res.update({'invoice_state': procurement.rule_id.invoice_state or procurement.invoice_state or 'none'})
+        invoice_state = 'none'
+        if procurement.invoice_state:
+            invoice_state = procurement.invoice_state
+        elif rocurement.rule_id.invoice_state:
```
Comment (pedrobaeza) [inline addons/stock_account/stock.py]: This should be 'procurement'

File: addons/stock_account/stock.py [hunk 2]
```diff
@@ -69,7 +69,12 @@ class procurement_order(osv.osv):
...
+            invoice_state = rocurement.rule_id.invoice_state
+        res.update({'invoice_state': invoice_state})
```
Comment (pedrobaeza) [inline addons/stock_account/stock.py:77]: You can write directly `res['invoice_state'] = invoice_state`

**Comment by pedrobaeza** [comment]
Ignacio, have you proposed this upstream on odoo?

File: addons/stock_account/stock.py (hunk 2 above)
Comment (acysos) [inline addons/stock_account/stock.py:77]: I know that I can do this. My first solution was this, but it ignore rule_id, so I write this solution.

**Comment by acysos** [comment]
I proposed other bug to Odoo, but they never reply. So I try it with OCA. I can try again.

**Comment by pedrobaeza** [comment]
Yeah, but OCB rules indicate that you must provide the same patch to odoo, to give the opportunity to be merged. Then, this can be taken into consideration due to:
- Odoo doesn't merge the upstream patch.
- They diverge from our opinion.
- Or they are very late.

File: addons/stock_account/stock.py [hunk 3]
```diff
@@ -69,7 +69,12 @@ class procurement_order(osv.osv):
 
     def _run_move_create(self, cr, uid, procurement, context=None):
         res = super(procurement_order, self)._run_move_create(cr, uid, procurement, context=context)
-        res.update({'invoice_state': procurement.rule_id.invoice_state or procurement.invoice_state or 'none'})
+        invoice_state = 'none'
+        if procurement.invoice_state:
+            invoice_state = procurement.invoice_state
+        elif procurement.rule_id.invoice_state:
+            invoice_state = rocurement.rule_id.invoice_state
```
Comment (pedrobaeza) [inline addons/stock_account/stock.py:76]: This is still bad. s/rocurement/procurement

**Comment by pedrobaeza** [comment]
I prefer this construction for your solution (that is basically to swap preference for the invoice_state flag):

```
res['invoice_state'] = procurement.invoice_state or procurement.rule_id.invoice_state or 'none'
```

**Comment by acysos** [comment]
Yes, this is other solution that I try:
res.update({'invoice_state': procurement.invoice_state or procurement.rule_id.invoice_state or  'none'})

Ok, I close the PR, I try again with Odoo. If they don't reply I request again the PR to OCA.

**Comment by pedrobaeza** [comment]
No, this shouldn't be exclusive. Typical process is propose in both places. Meanwhile Odoo discusses if the patch is merged or not, here in OCB we can merge it if we decide so. The requirement of making the PR upstream is to assure giving the possibility to Odoo of including in core the patches.

**Comment by pedrobaeza** [comment]
You can open this again meanwhile. Please put PR reference for upstream.

**Pull request closed without merging**